framework/include/interfaces/PerfGraphInterface.h

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef PERFGRAPHINTERFACE_H
+#define PERFGRAPHINTERFACE_H
+
+#include "Moose.h"
+#include "PerfGuard.h"
+#include "InputParameters.h"
+
+#define TIME_SECTION(id) PerfGuard time_guard(_perf_graph, id);
+
+// Forward declarations
+class PerfGraphInterface;
+
+template <>
+InputParameters validParams<PerfGraphInterface>();
+
+/**
+ * Interface for objects that needs transient capabilities
+ */
+class PerfGraphInterface
+{
+public:
+  /**
+   * For objects that _are_ MooseObjects
+   */
+  PerfGraphInterface(const MooseObject * moose_object, std::string prefix = "");
+
+  /**
+   * For objects that aren't MooseObjects
+   */
+  PerfGraphInterface(PerfGraph & perf_graph, std::string prefix = "");
+
+  virtual ~PerfGraphInterface();
```

Comment:
Suggestion: In C++11 we have the ability to just say `= default`, which I've been using in MOOSE lately. No need to define an empty destructor in the .C file anymore.

Reply:
👍